### PR TITLE
fix(dual-replay): report no aleatoric control under the reservoir policy

### LINKED ISSUE #363: Dual-replay reservoir writes report a fabricated aleatoric veto (available=True, passed=False) on every retained candidate
## Summary

`ReplayWriteResult` (`alberta_framework/core/dual_replay.py`) is documented as the "auditable long-term selection decision". Under `long_term_policy="reservoir"` no aleatoric control exists, and the priority diagnostics are correctly neutralized (`long_term_priority=0.0` with `long_term_priority_available=False`, components zeroed). The aleatoric pair is not: the reservoir branch sets `aleatoric_control_available = transition.aleatoric_uncertainty_available` (the *signal's* availability) while hard-coding `aleatoric_control_passed = False`. Every reservoir write with an available aleatoric signal therefore reads as "the control ran and vetoed this candidate" (`available & ~passed`), including writes where `wrote_long_term=True` on the same row.

Any reservoir-vs-calibrated A/B that tabulates noisy-TV veto rates from the write records gets 100% vetoes for the control arm on a stream where the aleatoric cap was never approached. No test covers these flags under reservoir (`tests/test_dual_replay.py` asserts them only on calibrated memories).

## Falsifier (deterministic, pristine `main` @ `45279f3`)

```python
import jax.numpy as jnp, jax.random as jr
from alberta_framework.core.dual_replay import DualReplayConfig, DualReplayMemory, ReplayOutcome, ReplayPrediction
f32 = lambda v: jnp.asarray(v, dtype=jnp.float32)
mem = DualReplayMemory(DualReplayConfig(total_capacity=6, short_term_capacity=3, observation_dim=2, action_dim=2,
    short_term_sample_size=2, long_term_sample_size=2, long_term_policy="reservoir", max_aleatoric_uncertainty=1.0))
state = mem.init(jr.key(0))
for p in range(1, 7):
    pred = ReplayPrediction(observation=f32([p, p+1]), action=f32([p % 2, 1.]),
        old_behavior_probability=f32(0.25), old_behavior_probability_available=jnp.asarray(True),
        old_behavior_logit=f32(-1.25), old_behavior_logit_available=jnp.asarray(True),
        old_value_target=f32(p), old_value_target_available=jnp.asarray(True),
        epistemic_surprise=f32(0.5), epistemic_surprise_available=jnp.asarray(True),
        aleatoric_uncertainty=f32(0.0), aleatoric_uncertainty_available=jnp.asarray(True),   # far below the cap
        representation_version=jnp.asarray(0, jnp.int32), provenance_id=jnp.asarray(p, jnp.int32),
        source_id=jnp.asarray(7, jnp.int32), valid=jnp.asarray(True))
    out = ReplayOutcome(next_observation=f32([p+1, p+2]), reward=f32(0.1), discount=f32(0.9),
        terminated=jnp.asarray(False), truncated=jnp.asarray(False), learning_progress=f32(0.5),
        learning_progress_available=jnp.asarray(True), safety_cost=f32(0.), safety_cost_available=jnp.asarray(True),
        valid=jnp.asarray(True))
    r = mem.record(state, pred, out); state = r.state
    print(p, "available", bool(r.aleatoric_control_available), "passed", bool(r.aleatoric_control_passed), "wrote_long", bool(r.wrote_long_term))
```

Output on `main`:

```
1 available True passed False wrote_long True
2 available True passed False wrote_long True
3 available True passed False wrote_long True
4 available True passed False wrote_long False
5 available True passed False wrote_long True
6 available True passed False wrote_long False
```

Expected: `aleatoric_control_available=False` under reservoir (no control was applied), exactly as `long_term_priority_available` already reads.

## Dedup

Related open (mine, different code path in the same module — calibrated-priority range/config validation): #313/#315; merge-tree composes cleanly. `gh issue/pr list --search "aleatoric"` / `"reservoir"`: nothing else.

Fix PR follows (failing-test-first).

### PR DESCRIPTION
**Pull request by Prabhat1308**
Closes #363.

Under long_term_policy="reservoir" no aleatoric control is applied, but the write receipt copied the signal-availability bit while hard-coding passed=False. That fabricated an aleatoric veto on every row with an available signal, including retained candidates.

The reservoir branch now reports both aleatoric_control_available=False and aleatoric_control_passed=False, matching its already-neutral priority diagnostics. Selection, RNG, state updates, and the calibrated branch are unchanged. The regression exercises low/high aleatoric values with both available and unavailable signal metadata.

Verification at 363ef20ed5df2be459e3ee3db17baccc825695fd:

- Rebased cleanly after merged #315 float32 priority/config hardening.
- 156 dual-replay tests passed under -W error.
- Ruff passed on changed source/tests.
- Strict mypy passed on dual_replay.py.
- outputs/ is untouched; no benchmark or seed lane was executed.

This is receipt/reporting correctness only and creates no scientific evidence.

- [x] logs: verification results inline above

### CODE REVIEW

**Review by lalalune — APPROVED** [review]
Independent review at exact head 112324d: READY. Reservoir admission has no aleatoric control, so reporting both availability and pass as false is the truthful diagnostic; the calibrated policy path remains unchanged. Verified with the current dual-replay stack: 156 tests passed, full Ruff and strict mypy passed, and the merge composition is clean. No registered evidence source or output artifact changes.
